tests/tools/app_store_connect/test_publish_action.py

```diff
@@ -65,21 +87,62 @@ def test_publish_action_with_username_and_password(_mock_altool, namespace_kwarg
         mock_get_packages.assert_called_with(patterns)
 
 
+def test_publish_action_with_localization_and_no_testflight_submission(publishing_namespace_kwargs, cli_argument_group):
+    BuildArgument.WHATS_NEW.register(cli_argument_group)
+    cli_args = argparse.Namespace(**publishing_namespace_kwargs)
+    patterns = [pathlib.Path('path.pattern')]
+    locale = Locale('en-GB')
+
+    with pytest.raises(argparse.ArgumentError) as error_info:
+        AppStoreConnect.from_cli_args(cli_args).publish(
+            application_package_path_patterns=patterns,
+            apple_id='name@example.com',
+            locale=locale,
+            whats_new=Types.WhatsNewArgument("What's new"),
+        )
+
+    assert str(error_info.value) == 'argument --whats-new: --testflight is required for submitting notes'
+
+
+def test_publish_action_testflight_with_localization(publishing_namespace_kwargs):
+    cli_args = argparse.Namespace(**publishing_namespace_kwargs)
+    with mock.patch.object(AppStoreConnect, 'find_paths') as mock_find_paths, \
+            mock.patch.object(AppStoreConnect, '_get_publishing_application_packages') as mock_get_packages, \
+            mock.patch.object(AppStoreConnect, '_upload_artifact_with_altool') as mock_upload, \
+            mock.patch.object(AppStoreConnect, '_validate_artifact_with_altool') as mock_validate, \
+            mock.patch.object(AppStoreConnect, '_get_uploaded_build') as mock_get_build, \
+            mock.patch.object(AppStoreConnect, 'create_beta_app_review_submission') as mock_create_review, \
+            mock.patch.object(AppStoreConnect, 'create_beta_build_localization') as mock_create_localization:
+        ipa_path = pathlib.Path('app.ipa')
+        mock_find_paths.return_value = [ipa_path]
+        mock_get_packages.return_value = [MockIpa(ipa_path)]
```

Comment:
I think it is better to get rid of the `MockIpa` class altogether as maintaining mirroring API in tests is too cumbersome. It is better to use proper mocking tools that are available in the stdlib. We can get the same results by using `create_autospec` from `unitest.mock` module.
```suggestion
        mock_get_packages.return_value = [create_autospec(Ipa, instance=True, path=ipa_path)]
```
Then the same thing can be used in the other places too where `MockIpa` is currently referenced.